### ISSUE TITLE
Kali Linux issue

### ISSUE THREAD

**Issue by fabianmendes** [Botspot/pi-apps#622]
Hi, I've tried to install it and it says something about authentication issue (see screenshot below)
![Screenshot_2021-05-01_09_58_52](https://user-images.githubusercontent.com/31230743/116781173-d6b24b00-aa70-11eb-813a-c811fb37351b.png)

**Comment by CleanMachine1** [comment]
Pi-Apps is made for Raspberry Pi OS, so expect errors.

I have a question for you though, why are you using Kali Linux at all?
I hope you understand that its not made to be used a Desktop Linux Distro.
So unless you have a need for kali (Please give me a reason if you do), switch as it doesn't make you look like a hacker, just makes you look inexperienced if you are using it as a desktop

**Comment by Botspot** [comment]
> Hi, I've tried to install **any app** and it says something about authentication issue (see screenshot below)
> ![Screenshot_2021-05-01_09_58_52](https://user-images.githubusercontent.com/31230743/116781173-d6b24b00-aa70-11eb-813a-c811fb37351b.png)

You can ignore those "No such file or directory" errors. They are hamless.
@fabianmendes When you try to install an app, does any terminal window appear?

**Comment by fabianmendes** [comment]
> > Hi, I've tried to install **any app** and it says something about authentication issue (see screenshot below)
> > ![Screenshot_2021-05-01_09_58_52](https://user-images.githubusercontent.com/31230743/116781173-d6b24b00-aa70-11eb-813a-c811fb37351b.png)
> 
> You can ignore those "No such file or directory" errors. They are hamless.
> @fabianmendes When you try to install an app, does any terminal window appear?

not from the terminal as you see, also I can remember that GUI just doesn't do anything when click install, it doesn't open the terminal.

**Comment by Botspot** [comment]
> not from the terminal as you see, also I can remember that GUI just doesn't do anything when click install, it doesn't open the terminal.

Hmm, sounds like another classic terminal issue. Is your Pi-Apps up to date? (downloaded recently, or months ago?)
@fabianmendes, what terminal(s) does Kali have installed? What command is required to run the terminal?

**Comment by fabianmendes** [comment]
> > not from the terminal as you see, also I can remember that GUI just doesn't do anything when click install, it doesn't open the terminal.
> 
> Hmm, sounds like another classic terminal issue. Is your Pi-Apps up to date? (downloaded recently, or months ago?)
> @fabianmendes, what terminal(s) does Kali have installed? What command is required to run the terminal?

Yes, It's on date. I've installed it recent (max. A day ago from this issue Open.
Terminal command: `exo-open --launch TerminalEmulator`
![Screenshot_2021-05-03_19_04_54](https://user-images.githubusercontent.com/31230743/116920687-7d017a80-ac42-11eb-9ae8-306932a9b76c.png)

I'm attaching a short video reproducing all. Thanks!

https://user-images.githubusercontent.com/31230743/116920840-a9b59200-ac42-11eb-99ed-9ed82fd2fc94.mp4
^ I hope it does help.

**Comment by Botspot** [comment]
> Yes, It's on date. I've installed it recent (max. A day ago from this issue Open.
> Terminal command: `exo-open --launch TerminalEmulator`

@fabianmendes What happens when you run this command?
```
xfce4-terminal
```

**Comment by fabianmendes** [comment]
> > Yes, It's on date. I've installed it recent (max. A day ago from this issue Open.
> > Terminal command: `exo-open --launch TerminalEmulator`
> 
> @fabianmendes What happens when you run this command?
> 
> ```
> xfce4-terminal
> ```

it asks me to install it. Also, the app don't let me install anything via gui either (i comment this 'cause you've changed the title)

![Screenshot 2021-05-08 22:36:53](https://user-images.githubusercontent.com/31230743/117555348-f4be1380-b04d-11eb-920e-0a80ce5e06f3.png)

**Comment by Botspot** [comment]
> it asks me to install it. Also, the app don't let me install anything via gui either (i comment this 'cause you've changed the title)

A working terminal is required for Pi-Apps to install any apps. The core of this issue is that the Pi-Apps `terminal-run` script didn't detect any working terminal emulators.

@fabianmendes, first, please don't install xfce4-terminal. While doing so would fix Pi-Apps, it would leave this problem intact for the next Kali Linux user. IUt's better to fix the issue, not work around it. :)
@fabianmendes what happens when you run [this command](https://www.geeksforgeeks.org/kali-linux-terminal-and-shell/#:~:text=So%20being%20one%20of%20the,environment%20is%20Xfce%2FXfce%20server.)?
```
qterminal
```

**Comment by CleanMachine1** [comment]
I really still want to know why you are using Kali?

**Comment by Botspot** [comment]
> I really still want to know why you are using Kali?

Personal preference. Just like that guy who was using Arch.
Pi-Apps does its best to accommodate as many systems and setups as possible.

**Comment by CleanMachine1** [comment]
I understand however Kali is a really unstable Desktop OS, I personally would only use it in VM or Live USB

**Comment by fabianmendes** [comment]
> > it asks me to install it. Also, the app don't let me install anything via gui either (i comment this 'cause you've changed the title)
> 
> A working terminal is required for Pi-Apps to install any apps. The core of this issue is that the Pi-Apps `terminal-run` script didn't detect any working terminal emulators.
> 
> @fabianmendes, first, please don't install xfce4-terminal. While doing so would fix Pi-Apps, it would leave this problem intact for the next Kali Linux user. It's better to fix the core issue, than to work around it. :)
> @fabianmendes what happens when you run [this command](https://www.geeksforgeeks.org/kali-linux-terminal-and-shell/#:~:text=So%20being%20one%20of%20the,environment%20is%20Xfce%2FXfce%20server.)?
> 
> ```
> qterminal
> ```

it did somwthing

![Screenshot 2021-05-09 03:35:53](https://user-images.githubusercontent.com/31230743/117559779-b1789a80-b077-11eb-94a4-5c080c521b65.png)

**Comment by Botspot** [comment]
> it did somwthing

Based on your screenshot, I've determined that Kali Linux uses `qterminal` by default.
As of commit d719c55bb9c9b61d02f229ef56232684459cb52c, Pi-apps now supports `qterminal`.

Thanks for reporting this issue so it could be fixed! :)
